src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Attributes/JsonNamingPolicyAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// When placed on a type, property, or field, indicates what <see cref="JsonNamingPolicy"/>
+    /// should be used when serializing or deserializing properties.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Interface | AttributeTargets.Struct | AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
+    public class JsonNamingPolicyAttribute : JsonAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonNamingPolicyAttribute"/>.
+        /// </summary>
+        public JsonNamingPolicyAttribute(JsonKnownNamingPolicy namingPolicy)
+        {
+            NamingPolicy = namingPolicy switch
+            {
+                JsonKnownNamingPolicy.CamelCase => JsonNamingPolicy.CamelCase,
+                JsonKnownNamingPolicy.SnakeCaseLower => JsonNamingPolicy.SnakeCaseLower,
+                JsonKnownNamingPolicy.SnakeCaseUpper => JsonNamingPolicy.SnakeCaseUpper,
+                JsonKnownNamingPolicy.KebabCaseLower => JsonNamingPolicy.KebabCaseLower,
+                _ => JsonNamingPolicy.CamelCase,
```

Comment:
This doesn't feel right to me, an invalid policy should probably result in no changes being made. It should probably also trigger a diagnostic if using the source generator.